### ISSUE TITLE
[a11y]: Multiple elements to be selected by using Modifier Keys 

### ISSUE THREAD

**Issue by psurana2** [carbon-design-system/carbon#16198]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Package version

v1.50.0

### React version

v17.0.2

### Automated testing tool and ruleset

Manual

### Description

Standard keyboard interaction for a component should be supported(the keys used for interaction are Escape, Spacebar, Enter, Home, End, PageUp, PageDown and the arrow keys, Modifier keys such as Shift, Ctrl and Alt may be used in conjunction with these keys, such as when selection of multiple items needs to be supported.) 

### Reproduction/example

https://stackblitz.com/edit/github-i1icxx?file=src%2FApp.jsx

### Steps to reproduce

Allow to select multiple rows of the Data table by keyboard using modified keys

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by laurenmrice** [comment]
Ref duplicate issue #4084
